app/src/main/java/com/raulh82vlc/functionalkotlinarrow/ui/MainActivity.kt

```diff
@@ -16,14 +16,60 @@
 
 package com.raulh82vlc.functionalkotlinarrow.ui
 
-import android.support.v7.app.AppCompatActivity
 import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import android.support.v7.widget.LinearLayoutManager
+import android.view.View
+import android.widget.Toast
+import arrow.effects.fix
 import com.raulh82vlc.functionalkotlinarrow.R
+import com.raulh82vlc.functionalkotlinarrow.data.cache.model.FeedItemCacheModel
+import com.raulh82vlc.functionalkotlinarrow.presentation.FeedListView
+import com.raulh82vlc.functionalkotlinarrow.presentation.Presentation
+import com.raulh82vlc.functionalkotlinarrow.ui.adapter.FeedAdapter
+import kotlinx.android.synthetic.main.activity_main.*
 
-class MainActivity : AppCompatActivity() {
+class MainActivity : AppCompatActivity(), FeedListView {
+    private lateinit var feedAdapter: FeedAdapter
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+        setFeedList()
```

Comment:
Named like a setter but doesn't have an argument. Makes it confusing. I would call it either `setupFeedList` or `initFeedList`

Reply:
yep, agreed